### ISSUE TITLE
No appindicator3 namespace on Ubuntu 17.10 (and no gnome shell extension support)

### ISSUE THREAD

**Issue by x2eliah** [stockmind/gpd-pocket-screen-indicator#5]
The screen-rotate indicator is not showing on an Ubuntu 17.10 (latest beta) respin. Using the default session. Gnome shell version is 3.26.1, however there is no installation support for anything on "extensions.gnome.org" (therefore no support for topicons). In theory the kAppIndicator should be installed and running, but doesn't seem to work either way.

Since beta2 is very close to final, this bug will likely affect the release ubuntu 17.10 users too.

Running  'gpdscreen-indicator' gives the following error:
Traceback (most recent last call):
  File "/usr/local/sbin/gpdscreen-indicator", line 19, in <module>
    gi.require_version('AppIndicator3', '0.1')
  File "/usr/lib/python2.7/dist-packages/gi/__init__.py", line 118, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace AppIndicator3 not available

.

other stats from problem-reporting.sh:
**Kernel:** 4.14.0-rc3-audio-gpdpocket
**Desktop Environment:**
**Display:**  :1 
**Monitor:** DSI1 connected primary 1920x1200+0+0 right (normal left inverted right x axis y axis) 95mm x 151mm
**Scripts:** gpdfand gpdscreen gpdscreen-indicator
**EFI:** [ 0.0000000] efi: EFI v2.40 by American Megatrends
**EFI:** Release Date: 08/07/2017

**Comment by stockmind** [comment]
If you move your mouse on bottom left angle a menu pops out with the indicator icon?

http://www.omgubuntu.co.uk/wp-content/uploads/2016/07/top-icons-plus.jpg <http://www.omgubuntu.co.uk/wp-content/uploads/2016/07/top-icons-plus.jpg>

**Comment by x2eliah** [comment]
Unfortunately, no. The Ubuntu sidebar occupies that space and there doesn't seem to be any foldout space (as on other ubuntu gnome installs. (I will attach a photo after an hour or so.)

**Comment by stockmind** [comment]
Thanks! I will have a look at it and will try on a virtual machine!

**Comment by x2eliah** [comment]
Here's how the desktop looks like. No old-style rollout and no topicon...
![screenshot from 2017-10-12 13-37-52](https://user-images.githubusercontent.com/32734282/31494244-bc2e83d8-af52-11e7-818f-bb56237e228e.png)

**Comment by stockmind** [comment]
I will look for a solution 💪

Il 12 ott 2017 13:39, "x2eliah" <notifications@github.com> ha scritto:

Here's how the desktop looks like. No old-style rollout and no topicon...
[image: screenshot from 2017-10-12 13-37-52]
<https://user-images.githubusercontent.com/32734282/31494244-bc2e83d8-af52-11e7-818f-bb56237e228e.png>

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/stockmind/gpd-pocket-screen-indicator/issues/5#issuecomment-336101999>,
or mute the thread

**Comment by stockmind** [comment]
Sorry had a better look at your issue report
What output give you this?

    sudo apt install gir1.2-appindicator3-0.1

After that you are able to launch the appindicator?

**Comment by x2eliah** [comment]
Hi, 
yes, the gpd-screen indicator is now launchable and shows up on topbar. (ofc it stayls alive only as long as the terminal is left on, but now this can be added to a startup script..). But, yes, this seems to solve it, thank you :)

**Comment by stockmind** [comment]
I need to check the dep install, odd! You were missing that dep, if you reboot now it should work and launch on startup! If not try to add it to startup application and let me know if you had to add it as a new entry to make it work! Thanks! :)
Also how you respin the 17.10 iso? Docker or standard way? 
Thanks!

**Comment by x2eliah** [comment]
The standard way, following the respin readme file and using the official 17.10 beta2 iso. With the 'gnome' flags.
(There were some squashfs errors at the nd of install, but no functional problems...)

Hmm, reboot alone did not help. (will do startup step soon-ish)

Edit: Okay, sorry, another reboot and it works without needing to add to startup scripts manually. So it was the missing dependency issue. Now everythings fine!

**Comment by x2eliah** [comment]
Appears to be fixed (checked with new respin of 17.10).

**Comment by Awkee** [comment]
My problems is :1234: 
```
Traceback (most recent call last):
  File "/home/zioer/.nutstore/dist/bin/nutstore-pydaemon.py", line 27, in <module>
    gi.require_version('AppIndicator3', '0.1')
  File "/usr/lib64/python3.6/site-packages/gi/__init__.py", line 130, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace AppIndicator3 not available

```
I solved this problems on my SUSE system by this :
```
## find the 
zypper search AppIndicator3

sudo zypper install typelib-1_0-AppIndicator3-0_1
```
